internal/providers/provider.go

```diff
@@ -385,6 +393,37 @@ type ImportedResource struct {
 	Private []byte
 }
 
+type MoveResourceStateRequest struct {
+	// SourceProviderAddress is the address of the provider that the resource
+	// is being moved from.
```

Comment:
Nit: Might be good to include these field-level comments in the protocol definition so it is more self-documenting. 👍

Reply:
Done!